core/commands/cat.go

```diff
@@ -20,36 +25,60 @@ it contains.
 	Arguments: []cmds.Argument{
 		cmds.StringArg("ipfs-path", true, true, "The path to the IPFS object(s) to be outputted").EnableStdin(),
 	},
-	Run: func(req cmds.Request) (interface{}, error) {
+	Run: func(req cmds.Request, res cmds.Response) {
 		node, err := req.Context().GetNode()
 		if err != nil {
-			return nil, err
+			res.SetError(err, cmds.ErrNormal)
+			return
 		}
 
 		readers := make([]io.Reader, 0, len(req.Arguments()))
 
-		readers, err = cat(node, req.Arguments())
+		readers, length, err := cat(node, req.Arguments())
 		if err != nil {
-			return nil, err
+			res.SetError(err, cmds.ErrNormal)
+			return
 		}
 
+		res.SetLength(length)
+
 		reader := io.MultiReader(readers...)
-		return reader, nil
+		res.SetOutput(reader)
+	},
+	PostRun: func(res cmds.Response) {
+		if res.Length() < progressBarMinSize {
+			return
+		}
+
+		bar := pb.New(int(res.Length())).SetUnits(pb.U_BYTES)
+		bar.Output = os.Stderr
+		bar.Start()
```

Comment:
i think this can probably be generalized rather than forcing the user to do it every time.
Also, the `os.Stderr` here is sketchy-- the command now is aware of the os. i think something like this makes more sense:

```go
bar.Output = res.Stderr
```

and inside the cmds lib:

```
res.Stdout = os.Stdout
res.Stderr = os.Stderr
```